monomer_test.go

```diff
@@ -1,19 +1,176 @@
 package monomer_test
 
 import (
+	"math/big"
+	"strconv"
 	"testing"
+	"time"
 
+	"github.com/cometbft/cometbft/types"
+	opeth "github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/beacon/engine"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/polymerdao/monomer"
 	"github.com/polymerdao/monomer/testutils"
 	"github.com/stretchr/testify/require"
 )
 
-func TestBlockToEth(t *testing.T) {
-	l1InfoTx, depositTx, cosmosEthTx := testutils.GenerateEthTxs(t)
-	block := testutils.GenerateBlockFromEthTxs(t, l1InfoTx, []*ethtypes.Transaction{depositTx}, []*ethtypes.Transaction{cosmosEthTx})
-	ethBlock, err := block.ToEth()
-	require.NoError(t, err)
-	for i, ethTx := range []*ethtypes.Transaction{l1InfoTx, depositTx, cosmosEthTx} {
-		require.EqualExportedValues(t, ethTx, ethBlock.Body().Transactions[i])
+func TestChainID(t *testing.T) {
+	n := int64(12345)
+	id := monomer.ChainID(n)
+
+	idStr := strconv.FormatInt(n, 10)
+	require.Equal(t, idStr, id.String())
+
+	bigInt := big.NewInt(n)
+	require.Equal(t, bigInt, id.Big())
+
+	hexID := (*hexutil.Big)(bigInt)
+	require.Equal(t, hexID, id.HexBig())
+}
+
+func newTestHeader(timestamp uint64) *monomer.Header {
+	return &monomer.Header{
+		ChainID:    12345,
+		Height:     67890,
+		Time:       timestamp,
+		StateRoot:  common.HexToHash("0x1"),
+		ParentHash: common.HexToHash("0x2"),
+		GasLimit:   3000000,
+		Hash:       common.HexToHash("0x3"),
 	}
 }
+
+func TestToComet(t *testing.T) {
+	timestamp := time.Now()
+	header := newTestHeader(uint64(timestamp.Unix()))
+	cometHeader := header.ToComet()
+
+	require.Equal(t, header.ChainID.String(), cometHeader.ChainID)
+	require.Equal(t, header.Height, cometHeader.Height)
+	require.Equal(t, timestamp.Unix(), cometHeader.Time.Unix())
+	require.Equal(t, header.StateRoot.Bytes(), cometHeader.AppHash.Bytes())
```

Comment:
It is better to require equality on the entire struct. This way when fields are added, the test will fail, warning us of a potential issue. Something like:
```suggestion
	require.Equal(t, &cometbft.Header{
		ChainID: header.ChainID.String(),
		Height: header.Height,
		// the rest of the fields.
	}, cometHeader)
```